### DISCUSSION TITLE
Watch history by update timestamp

### DISCUSSION THREAD

**Discussion by mwrod** [trakt/trakt-api#317] [Q&A]
**My Use Case (Daily Sync)**: I want to get all new episode watches, which were marked as watched in a given time frame (in my case all episode watches of the last 24 hours). What's important here is that I don't want to get episodes watches with watch dates in a given time frame (not `watched_at` date but `last_updated_at` timestamp!).

**Solution (with the current API):** Retrieve all shows which were updated in a given time frame by [Get Watched Endpoint](https://trakt.docs.apiary.io/#reference/sync/get-watched/get-watched). After that I would get the full history of each show with the [Get watched history endpoint](https://trakt.docs.apiary.io/#reference/sync/get-history/get-watched-history). Potentially I need to check each page for new episode watches because I can not filter the result by `last_updated_at` timestamp and they are sorted by `watched_at` timestamp.

**Suggestion**: Either add a new endpoint to retrieve watched history by `last_updated_at` timestamp or add an option to sort the [Get watched history](https://trakt.docs.apiary.io/#reference/users/history/get-watched-history) by `last_updated_at` timestamp.

I would think that syncing watches is used by many applications. By the above suggestion the load would be reduced and the API would be easier to use for syncing processes. What do you think? Is it a reasonable extension to the API? Or do I maybe miss something in the API?

**Comment by rectifyer** [comment]
Can’t you just use the [Get Watched Endpoint](https://trakt.docs.apiary.io/#reference/sync/get-watched/get-watched) only to do this? Since this method has the entire watched history, there is no need to get history separately. Since you’re caching data in your app, you’ll know when new episodes show up in the [Get Watched Endpoint](https://trakt.docs.apiary.io/#reference/sync/get-watched/get-watched).

Another reason for using this endpoint is you can also handle when a user removes any plays and fully sync the state of all shows.

**Reply by mwrod** [reply]
Thanks for your reply @rectifyer. You are right, `Get Watched Endpoint` seems to have enough information for my purpose!

One downside of using the endpoint is that it provides the full history of the user. I measured the network traffic with a user with ~50k episode watches and it was approx. 1/2 MB. It would be great to to have one of the following two options:
- Get Watched endpoint as a paginated endpoint, which is sorted decrementing by `last_update_timestamp`
- Get Watched endpoint with a parameter "last_update_start_at" in order to have only the new updates

Would you consider adding either of them in your API?
